### PR TITLE
Remove duplication for Descriptable implementations using default methods

### PR DESCRIPTION
**Pull request by kriegfrj**
Just an idea I had. A precursor for a bigger idea.

### CODE REVIEW

File: src/main/java/org/assertj/core/api/Descriptable.java [hunk 1]
```diff
@@ -23,7 +24,7 @@
  * @author Yvonne Wang
  * @author Mikhail Mazursky
  */
-public interface Descriptable<SELF> {
+public interface Descriptable<SELF extends Descriptable<SELF>> {
```
Comment (filiphr) [inline src/main/java/org/assertj/core/api/Descriptable.java]: Just FYI this change is not a backwards compatible change. Why is it needed to force that `SELF` extends from `Descriptable`?

Comment (kriegfrj) [reply src/main/java/org/assertj/core/api/Descriptable.java]: This is true. However, it's not really a "SELF" type if it doesn't extend itself? I think it was probably an oversight when the type was originally created - I can't think of when you'd have a use case that SELF is not an instance of `Descriptable`.

On the other hand, it is not essential to the main purpose of this PR, so I can leave this change out of this PR for further discussion. The rest of the PR should (hopefully!) be less controversial.

Comment (joel-costigliola) [reply src/main/java/org/assertj/core/api/Descriptable.java]: Let's take the safest approach and not extend from Descriptable.

Comment (kriegfrj) [reply src/main/java/org/assertj/core/api/Descriptable.java]: Latest push has removed the constraint on `SELF`.

**Comment by joel-costigliola** [comment]
Fixed in 618609cc0a05414ef1a117cf616621310cd062ee, thanks @kriegfrj.

**Pull request closed without merging**